I don't know how to use tui::widgets::Table effectively

The `Table` type is giving me some type-related difficulty when I try to use it in a practical application.

What I want to do is display a formatted (read: truncated and aligned) float as one of the columns in a table, e.g.

```
HEADER1  HEADER2      THING
-------  -------      -----
name     other-thing   1.99
name2    description  25.80
```

Sounds simple, right? Not so fast. Granted, I am relatively new to Rust, so I may be missing something.

`Row::Data(x)` expects `x` to be an `Iterator<D>`, where type `D` is `Display`. This means that every element must be the same type (no surprise) and that that type must implement `Display` (also no surprise). Okay, so that means I need to provide all of my row values as some sort of string type. Maybe like this?

```rust
let data_vec = vec![
        "name".to_string(),
        "other-thing".to_string(),
        format!("{:5.2}", 1.99)
    ];
let data_row = Row::Data(data_vec.into_iter());
```

Looks good so far. Now let's try to use it in a UI.

```rust
let size = term.size()?;
Group::default()
    .sizes(&[Size::Percent(100)])
    .render(term, &size, |t, c| {
        Table::new(
            vec!["HEADER1", "HEADER2", "THING"].into_iter(),
            vec![data_row].into_iter()
        ).block(Block::default())
            .render(t, &c[0]);
    });
```

Hold on, what?

```
error[E0507]: cannot move out of captured outer variable in an `FnMut` closure
  --> src/main.rs:92:22
   |
85 |     let data_row = Row::Data(data_vec.into_iter());
   |         -------- captured outer variable
...
92 |                 vec![data_row].into_iter()
   |                      ^^^^^^^^ cannot move out of captured outer variable in an `FnMut` closure
```

I think that this is something ownership-related. Maybe we should use references. Let's change the first block to look like this:

```rust
let formatted_flt = format!("{:5.2}", 1.99);
let data_vec = vec![
        "name",
        "other-thing",
        &formatted_flt
    ];
let data_row = Row::Data(data_vec.into_iter());
```

Nope.

```
error[E0507]: cannot move out of captured outer variable in an `FnMut` closure
  --> src/main.rs:94:22
   |
87 |     let data_row = Row::Data(data_vec.into_iter());
   |         -------- captured outer variable
...
94 |                 vec![data_row].into_iter()
   |                      ^^^^^^^^ cannot move out of captured outer variable in an `FnMut` closure
```

In the demos, this is avoided by using string literals as the ultimate source for the tabular data:

https://github.com/fdehau/tui-rs/blob/ef2054a45b03782a54ec85aa75b0a52e7793b61c/examples/table.rs#L25-L32

Likewise in the docs:

https://github.com/fdehau/tui-rs/blob/ef2054a45b03782a54ec85aa75b0a52e7793b61c/src/widgets/table.rs#L30-L35

I really hope that the problem here is my understanding of the Rust type system. If not, then this is not a useful implementation of a user-facing table (ususally, tables are used to display information that is created or discovered at runtime).

[my M(non-)WE with a .txt extension for Github's file uploader](https://github.com/fdehau/tui-rs/files/1976608/main.txt)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,10 @@
 
 * `Terminal` implements `Debug`
 
+### Changed
+
+* Use `FnOnce` instead of `FnMut` in Group::render
+
 ## v0.2.1 - 2018-04-01
 
 ### Added

--- a/examples/table.rs
+++ b/examples/table.rs
@@ -84,23 +84,23 @@ fn main() {
 }
 
 fn draw(t: &mut Terminal<MouseBackend>, app: &App) {
+    let selected_style = Style::default().fg(Color::Yellow).modifier(Modifier::Bold);
+    let normal_style = Style::default().fg(Color::White);
+    let header = ["Header1", "Header2", "Header3"];
+    let rows = app.items.iter().enumerate().map(|(i, item)| {
+        if i == app.selected {
+            Row::StyledData(item.into_iter(), &selected_style)
+        } else {
+            Row::StyledData(item.into_iter(), &normal_style)
+        }
+    });
     Group::default()
         .direction(Direction::Horizontal)
         .sizes(&[Size::Percent(100)])
         .margin(5)
         .render(t, &app.size, |t, chunks| {
-            let selected_style = Style::default().fg(Color::Yellow).modifier(Modifier::Bold);
-            let normal_style = Style::default().fg(Color::White);
-            Table::new(
-                ["Header1", "Header2", "Header3"].into_iter(),
-                app.items.iter().enumerate().map(|(i, item)| {
-                    if i == app.selected {
-                        Row::StyledData(item.into_iter(), &selected_style)
-                    } else {
-                        Row::StyledData(item.into_iter(), &normal_style)
-                    }
-                }),
-            ).block(Block::default().borders(Borders::ALL).title("Table"))
+            Table::new(header.into_iter(), rows)
+                .block(Block::default().borders(Borders::ALL).title("Table"))
                 .widths(&[10, 10, 10])
                 .render(t, &chunks[0]);
         });

--- a/src/layout.rs
+++ b/src/layout.rs
@@ -326,10 +326,10 @@ impl Group {
         self.sizes = Vec::from(sizes);
         self
     }
-    pub fn render<F, B>(&self, t: &mut Terminal<B>, area: &Rect, mut f: F)
+    pub fn render<F, B>(&self, t: &mut Terminal<B>, area: &Rect, f: F)
     where
         B: Backend,
-        F: FnMut(&mut Terminal<B>, &[Rect]),
+        F: FnOnce(&mut Terminal<B>, &[Rect]),
     {
         let chunks = t.compute_layout(self, area);
         f(t, &chunks);

